.integration/project-plugins-loader.js

```diff
@@ -0,0 +1,196 @@
+    path.lastIndexOf('/') + 1,
+    path.lastIndexOf('.')
+  );
+  return filename;
+};
+
+/**
+ * Loads asynchronously the script which is located at the given `path`.
+ * @param {String} `path` the script's path (under the `js/plugins` directory)
+ * @param {Function()} `loaded_callback` the function to invoke once the script has been loaded
+ */
+Project.load_script = path => loaded_callback => {
+  const url = `${PluginManager._path}/${path}`;
+  const node = document.createElement('script');
+  node.type = 'text/javascript';
+  node.src = url;
+  node.async = false;
+  node.onerror = loaded_callback;
+  node.onload = loaded_callback;
+  document.body.appendChild(node);
+};
+
+/**
+ * Short-hand to constructs a new plugin representation (as used in `plugins.list.js`).
+ * @param {String} `name` the plugin's short name
+ * @param {String} `source` the plugin's main script's location
+ * @param {Array<String>} `dependencies` the list of plugin's dependencies (paths to additional `.js` to load)
+ * @eturn the structure which holds one plugin's information.
+ */
+Project.Plugin = (name, source, dependencies = []) => {
+  return {
+    name: name,
+    source: source,
+    dependencies: dependencies,
+  };
+};
+
+// Plug-in's setup
+(function(context) {
+  // Parameters' formatter
+  const string = default_value => current_value => {
+    return (current_value || '').trim() || default_value;
+  };
+
+  // Parameters' retrieval
+  const parameters = PluginManager.parameters(Project.filename());
+  const plugins_dir = string('witch-workshop.plug-ins/plugins')(
+    parameters["Plugins' directory"]
+  );
+  const plugins_list_src = `${plugins_dir}/plugins.list.js`;
+
+  // Loaders
+  const load_main_script = plugin => callback_once_loaded => {
+    const src = `${plugins_dir}/${plugin.source}`;
+    console.log(
+      ...Project.log.format(
+        `Loading main script of project's plugin: ${plugin.name} (${src})`
+      )
+    );
+    Project.load_script(src)(callback_once_loaded);
+  };
+
+  const load_dependency = dependency_sub_path => callback_once_loaded => {
+    const src = `${plugins_dir}/${dependency_sub_path}`;
+    console.log(...Project.log.format(`Loading plugin's dependency: (${src})`));
+    Project.load_script(src)(callback_once_loaded);
+  };
+
+  const load_plugin = plugin => callback_once_loaded => {
+    return () => {
+      Project.log.group(`Loading project's plugin: ${plugin.name}`)(
+        end_plugin_group => {
+          Project.log.group(`Loading dependencies for: ${plugin.name}`)(
+            end_dependencies_group => {
+              plugin.dependencies.reverse().reduce(
+                (Σ, dependency_sub_path) =>
+                  load_dependency(dependency_sub_path)(Σ),
+                () => {
+                  end_dependencies_group();
+                  load_main_script(plugin)(() => {
+                    end_plugin_group();
+                    callback_once_loaded();
+                  });
```

Comment:
Fixed (_somehow_ 🤔) with: [#658b0d70502548d60c3cef92eb9628386bb07f8d](https://github.com/random-mode/witch-workshop.plug-ins/pull/3/commits/658b0d70502548d60c3cef92eb9628386bb07f8d).

However, I bet that you wanted me to use `Promise` instead of these illegible callbacks 😅, aren't you?